### PR TITLE
keys: improve keys.Addr error messages

### PR DESCRIPTION
**Pull request by bdarnell**
Keys can be non-addressable without being malformed.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5779)

### CODE REVIEW

File: keys/keys_test.go [hunk 1]
```diff
@@ -93,26 +93,40 @@ func TestKeyAddress(t *testing.T) {
...
+			RangeDescriptorKey(roachpb.RKey(RangeLastVerificationTimestampKey(0))),
+		},
+		"local key .* malformed": {
+			makeKey(localPrefix, roachpb.Key("z")),
+		},
 	}
-	for i, test := range testCases {
-		if addr, err := Addr(test); !testutils.IsError(err, "local key .* malformed; should contain prefix .*") {
-			t.Errorf("%d: expected addressing key %q to throw error, but it returned address %q", i, test, addr)
+	i := 0
```
Comment (tamird) [inline keys/keys_test.go]: `i`?

Comment (bdarnell) [inline keys/keys_test.go]: For the %d in the error message. But It looks like I forgot to actually increment it, and now that the keys pretty-print it's not as valuable. I'll just remove it.

**Comment by nvanbenschoten** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[keys/keys.go, line 314 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5779#-KEDKfC6x9XEe13e0udn:-KEDKfC6x9XEe13e0udo:1115721862) ([raw file](https://github.com/cockroachdb/cockroach/blob/cd05665f1d4345fcb92424bd6608c92e771edafc/keys/keys.go#L314)):*
I have a tendency to like switch-case statements once the number of conditions gets this large, but either way works!

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5779#-:-KEDL6fMUyhl0fks9bx_:1417088721)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[keys/keys_test.go, line 123 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5779#-KEDNMur2hMZbMWQfNd6:-KEDNMur2hMZbMWQfNd7:-658686958) ([raw file](https://github.com/cockroachdb/cockroach/blob/cd05665f1d4345fcb92424bd6608c92e771edafc/keys/keys_test.go#L123)):*
i think you should prefix these messages with `regexp` and `key`'s index.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5779#-:-KEDNTpG7kkTSfmhnn5q:1417088721)*

**Pull request merged**